core/cmd/hoverfly/main.go

```diff
@@ -179,6 +181,36 @@ func init() {
 	goproxy.GoproxyCa = tlsc
 }
 
+func removeDuplicates(elements []string) []string { // change string to int here if required
+	// Use map to record duplicates as we find them.
+	encountered := map[string]bool{} // change string to int here if required
+	result := []string{}             // change string to int here if required
+
+	for v := range elements {
+		if encountered[elements[v]] == true {
+			// Do not add duplicate.
+		} else {
+			// Record this element as an encountered element.
+			encountered[elements[v]] = true
+			// Append to result slice.
+			result = append(result, elements[v])
+		}
+	}
+	// Return the new slice.
+	return result
+}
+
+// Find takes a slice and looks for an element in it. If found it will
+// return it's key, otherwise it will return -1 and a bool of false.
+func Find(slice []string, val string) (int, bool) {
+	for i, item := range slice {
+		if item == val {
+			return i, true
+		}
+	}
+	return -1, false
+}
+
```

Comment:
I think putting the log output values to a map would simplify this: 
```
logOutputMap := map[string]string{}
for _, val  := range logOutputFlags {
	logOutputMap[val] = val
}
```
the map key is unique, so it's automatically de-duplicate, and then you can just check if the map contains a key == `file`, something like this: 
```
if val, ok := logOutputMap["file"]; ok {
	//do something here
}
```

Reply:
Thank you for your suggestions. I updated source codes as you mentioned. It becomes simple and clean now.